compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -185,7 +185,17 @@ pub(crate) fn target_machine_factory(
     let reloc_model = to_llvm_relocation_model(sess.relocation_model());
 
     let (opt_level, _) = to_llvm_opt_settings(optlvl);
-    let use_softfp = sess.opts.cg.soft_float;
+    let use_softfp = if sess.target.arch == "arm" && sess.target.abi == "eabihf" {
+        sess.opts.cg.soft_float
+    } else {
+        // All `use_softfp` does is the equivalent of `-mfloat-abi` in GCC/clang, which only exists on ARM targets.
+        // We document this flag to only affect `*eabihf` targets, so let's show a warning for all other targets.
+        if sess.opts.cg.soft_float {
+            sess.dcx().emit_warn(errors::SoftFloatIgnored);
```

Comment:
Wouldn't it be better to have this logic out-side of cg_llvm (maybe in `rustc_session`) so that the other codegen backends have that warning "for free"?

Reply:
A lot of the target logic is in this crate, I don't plan to do all that refactoring.

Also, this file is currently literally the old file in the repo that even looks at the `soft_target` flag.

Reply:
Turns out we already have a `validate_commandline_args_with_session_available` function so fair, I moved the checks there.